internal/commands/runcommand.go

```diff
@@ -13,11 +15,15 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var (
+	logger = flume.New("runcommand")
+)
+
 func commandRunE(command Command, config *config.Config) func(cmd *cobra.Command, args []string) error {
 	switch typedCommand := command.(type) {
 	case NoArgumentCommand:
 		return func(cmd *cobra.Command, args []string) error {
-			log.Debug("runcommand.go", "commandRunE() NoArgumentCommand", "Cmd path:", cmd.CommandPath(), "Arguments:", args)
+			log.Debug(logger, cmd.CommandPath(), "Arguments:", args)
```

Comment:
Not a big fan of this signature, tbh. Maybe just change our internal logging package to be a wrapper around flume, if we really need this fine-grained control (I'm not convinced we do, as we really only have two scopes - the plumbing and the command being ran)? eg, something like;
```
type Logger struct {
   flume.Logger
}
func New(scope string) *Logger {
  return &Logger{ flume.New(scope) }
}
func (l *Logger) Debug(msg string, args ...interface{}) {
  l.Logger.Debug(msg,args)
}
func (l *Logger) Debug(msg string, args ...interface{}) {
  l.Logger.Debug(msg,args)
}
```
But then again, that raises the question of why not just use flume directly? if flume.Configure applies to all flume instances, I guess we could just implement the level logic somewhere like commmands and go with that?

Reply:
Yeah. So we could just in bootstrap.go declare the flume config like this?
``` golang
			flume.Configure(flume.Config{
				AddCaller:    &conf.GlobalFlags.Debug,
				DefaultLevel: flume.DebugLevel,
				Encoding:     "term-color",
			})
			if !conf.GlobalFlags.Debug {
				flume.Configure(flume.Config{
					AddCaller:    &conf.GlobalFlags.Debug,
					DefaultLevel: flume.InfoLevel,
					Encoding:     "term-color",
				})
			}
```

And just in the pckgs init flume and use it. I think you're right and should be enough for now at least.

Reply:
This way we also get the caller from flume right `@:runcommand@commands/runcommand.go:47`